server/schedule/hot_cache.go

```diff
@@ -131,10 +132,19 @@ func (w *HotSpotCache) isNeedUpdateStatCache(region *core.RegionInfo, flowBytes
 		Version:        region.GetRegionEpoch().GetVersion(),
 		AntiCount:      hotRegionAntiCount,
 	}
+	// notSameMagnitude to judge that two numbers are not in the same order of magnitude
+	notSameMagnitude := func(a, b float64) bool {
+		return math.Abs(math.Log10(a)-math.Log10(b)) >= 1
+	}
 
 	if oldItem != nil {
 		newItem.HotDegree = oldItem.HotDegree + 1
+		// denosing
+		if newItem.StoreID != oldItem.StoreID && notSameMagnitude(float64(oldItem.FlowBytes), float64(newItem.FlowBytes)) {
```

Comment:
I don't think it is a good idea.
For example, if a region's flowBytes are reported as '1, 10, 10, 10 ...', you will always use 1, which is apparently not correct.

Reply:
Maybe a simple median filter is better. See https://en.wikipedia.org/wiki/Median_filter

Reply:
this condition's precondition is leader change, that means the store id is not same. the normal report not yet been denosing now.